packages/core/src/shader/state/BlendState.ts

```diff
@@ -167,6 +172,11 @@ export class BlendState {
       colorWriteMask
     } = this.targetBlendState;
 
+    if (customStates) {
+      const colorWriteMaskState = customStates[RenderStateElementKey.BlendStateColorWriteMask0];
+      colorWriteMaskState !== undefined && (colorWriteMask = <number>colorWriteMaskState);
```

Comment:
Fix number to ColorWriteMask and check other funcs

Reply:
done